src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php

```diff
@@ -194,8 +204,12 @@ public function get(): ?array
             }
 
             // use SELECT ... FOR UPDATE to lock table
+            if (!method_exists(QueryBuilder::class, 'forUpdate')) {
```

Comment:
We cannot lock anymore in DBAL4?

Reply:
as far as I understand the related changes in Doctrine DBAL this now happens with the `forUpdate()` call above

Reply:
indeed